### PR TITLE
DashGPT: Disable GenAI title and description buttons for empty dashboards

### LINKED ISSUE #76986: Auto-generate: Do not allow to generate title and description for an empty dashboard
Disable the button with the tooltip explaining that panels should have titles / descriptions in order to generate useful dashboard title / descriptions

In future investigate possibility of being able to generate panel title / descriptions automatically as part of this flow vs just blocking the user and making the process more manual.

### PR DESCRIPTION
**Pull request by harisrozajac**
This will disable the genAI title and description  buttons when:
- the dashboard is empty
- the dashboard has panels but all panels: 
    - have no title or have "Panel title" as title AND
    - have no description

https://github.com/user-attachments/assets/8c042bc6-99aa-4b68-9291-7f8dff69e1d5

Fixes #76986

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.

### CODE REVIEW

File: public/app/features/dashboard/components/GenAI/GenAIDashDescriptionButton.tsx [hunk 1]
```diff
@@ -22,27 +24,29 @@ const DESCRIPTION_GENERATION_STANDARD_PROMPT =
...
-
+function getMessages(dashboard: DashboardModel): Message[] {
   return [
     {
       content: DESCRIPTION_GENERATION_STANDARD_PROMPT,
       role: Role.system,
     },
     {
-      content: `The title of the dashboard is "${dashboard.title}"\n` + `${panelPrompt}`,
+      content: `The title of the dashboard is "${dashboard.title}"\n` + `${getDashboardPanelPrompt(dashboard)}`,
```
Comment (harisrozajac) [inline public/app/features/dashboard/components/GenAI/GenAIDashDescriptionButton.tsx:49]: changed this just to be consistent with style in `GenAIDashTitleButton.tsx`

File: public/app/features/dashboard/components/GenAI/utils.ts [hunk 2]
```diff
@@ -158,3 +154,18 @@ export function getFilteredPanelString(panel: Panel): string {
 
   return JSON.stringify(filteredPanel, null, 2);
 }
+
+export function getPanelStrings(dashboard: DashboardModel): string[] {
+  const panelStrings = dashboard.panels
+    .filter(
+      (panel) =>
+        (panel.title.length > 0 && panel.title !== 'Panel title') || (panel.description && panel.description.length > 0)
```
Comment (harisrozajac) [inline public/app/features/dashboard/components/GenAI/utils.ts:162]: We could potentially add a check here for a description so if description is "description" or "panel description" then we filter it out. However, descriptions are not auto-created like titles.

**Review by nmarrs — COMMENTED** [review]
This is looking great and really cool to see this code being worked on again ❤️

**Review by nmarrs — APPROVED** [review]
Was energized so decided to address feedback I had for PR in latest commit 😬 

I think this is good to go 🚀 

@ivanortegaalba it would be great to get your review though as well if possible :) 

https://github.com/user-attachments/assets/e3b99217-bbd5-4a8f-90f2-301b23731109

**Review by ivanortegaalba — APPROVED** [review]
Great work here peeps! 🔥 I tested it, it works like a charm.

**Pull request merged**